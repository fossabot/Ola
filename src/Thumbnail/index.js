import React, { useRef } from 'react'
import { default as PT } from 'prop-types'
import cx from 'classnames'

const Thumbnail = ({ className, url, errorImage, borderColor, children }) => {
  const imgError = useRef(null)

  return (
    <div
      className={cx('ola_thumbnail', className)}
      style={{ '--color': borderColor }}
    >
      <img
        className={'ola_thumbnail-image'}
        src={url}
        onError={() => errorImage ? imgError.current.src = errorImage : null }
        ref={imgError}
      />
      {children}
    </div>
  )
}

export default Thumbnail

Thumbnail.defaultProps = {
  className: null,
  url: null,
  borderColor: null,
}

Thumbnail.propTypes = {
  /** Extra className */
  className: PT.string,
  /** Image url */
  url: PT.string,
  /** Error image */
  errorImage: PT.string,
  /** Color of the column */
  borderColor: PT.string,
  /** Childen nodes */
  children: PT.oneOfType([PT.string, PT.arrayOf(PT.node), PT.node]),
}
