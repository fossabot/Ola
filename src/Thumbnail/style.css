.ola_thumbnail {

}

.ola_thumbnail-image {
    border: var(--size-2) solid rgb(14, 185, 23);
    box-shadow: var(--shadow-2), var(--shadow-line);
    border-radius: var(--radius);
}