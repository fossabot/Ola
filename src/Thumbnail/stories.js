import React from 'react'
import Thumbnail from './'

export default {
  title: 'Thumbnail',
}

export const Base = (args) => <Thumbnail {...args} url="https://placehold.it/100x100"/>
export const Border = (args) => <div style={{ display: 'flex' }}>
  <Thumbnail {...args} borderColor={'#F48498'} url="https://placehold.it/100x100"/>
  <Thumbnail {...args} borderColor={'#7FD8BE'} url="https://placehold.it/100x100"/>
  <Thumbnail {...args} borderColor={'#FFD266'} url="https://placehold.it/100x100"/>
  <Thumbnail {...args} borderColor={'#95B8D1'} url="https://placehold.it/100x100"/>
</div>