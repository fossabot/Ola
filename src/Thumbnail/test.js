import React from 'react'
import Thumbnail from './'
import renderer from 'react-test-renderer'

it('Default Thumbnail', () => {
  const tree = renderer.create(<Thumbnail src="https://placehold.it/100x100" errorImage="https://via.placeholder.com/100/000000" />).toJSON()
  expect(tree).toMatchSnapshot()
})