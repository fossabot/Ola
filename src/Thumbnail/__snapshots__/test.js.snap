// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Default Thumbnail 1`] = `
<div
  className="ola_thumbnail"
  style={
    Object {
      "--color": null,
    }
  }
>
  <svg
    className="ola_thumbnail-image"
    height="300"
    viewBox="0 0 400 300"
    width="400"
  >
    <path
      d="M270 94a8 8 0 018 8v94a8 8 0 01-8 8H130a8 8 0 01-8-8v-94a8 8 0 018-8h140zm6 22H124v79a7 7 0 007 7h138a7 7 0 007-7v-79z"
    />
    <circle
      cx="140"
      cy="105"
      r="5"
    />
    <circle
      cx="159"
      cy="105"
      r="5"
    />
    <circle
      cx="178"
      cy="105"
      r="5"
    />
  </svg>
</div>
`;
